### DISCUSSION TITLE
Error in documentation regarding typescript and persisting store data

### DISCUSSION THREAD

**Discussion by alex-zwingli** [pmndrs/zustand#2736] [Bug report] [answered]
### Bug Description

The documentation [here](https://zustand.docs.pmnd.rs/integrations/persisting-store-data#how-do-i-use-it-with-typescript) says that you just have to switch `create(...)` with `create<State>()(...)`; however, I need to use `create<State, any>()(...)` to avoid highlighted errors. 

### Reproduction Link

//

**Comment by dbritto-dev** [comment]
@alex-zwingli what errors?

**Reply by alex-zwingli** [reply]
Red wavy lines in vs code saying that the types weren't compatible.

**Reply by dbritto-dev** [reply]
@alex-zwingli any ts playground?

**Reply by alex-zwingli** [reply]
``` typescript
import { create } from "zustand";
import { persist } from "zustand/middleware";

type State = {
  highlightedText: string[];
  highlightedTradePartnerId: string[];
  showSubPath: boolean;
  highlightToday: boolean;
};

type Action = {
  setHighlightedText: (text: string[]) => void;
  setHighlightedTradePartnerId: (tradePartnerId: string[]) => void;
  setShowSubPath: (show: boolean) => void;
  setHighlightToday: (highlight: boolean) => void;
};

export const useScheduleSettings = create<State & Action, any>(
  persist(
    (set) => ({
      highlightedText: [],
      setHighlightedText: (text) => set({ highlightedText: text }),

      highlightedTradePartnerId: [],
      setHighlightedTradePartnerId: (tradePartnerId) =>
        set({ highlightedTradePartnerId: tradePartnerId }),

      showSubPath: false,
      setShowSubPath: (show) => set({ showSubPath: show }),

      highlightToday: false,
      setHighlightToday: (highlight) => set({ highlightToday: highlight }),
    }),
    {
      name: "schedule-settings", // unique name for the localStorage key
    }
  )
);
```

**Reply by alex-zwingli** [reply]
This is the problem that vs code shows me if switch `<State & Action, any>` to <State & Action>`

```
Argument of type 'StateCreator<State & Action, [], [["zustand/persist", State & Action]]>' is not assignable to parameter of type 'StateCreator<State & Action, [], []>'.
  Type 'StateCreator<State & Action, [], [["zustand/persist", State & Action]]>' is not assignable to type '{ $$storeMutators?: [] | undefined; }'.
    Types of property '$$storeMutators' are incompatible.
      Type '[["zustand/persist", State & Action]] | undefined' is not assignable to type '[] | undefined'.
        Type '[["zustand/persist", State & Action]]' is not assignable to type '[]'.
          Source has 1 element(s) but target allows only 0.ts(2345)
```

**Reply by dbritto-dev** [reply, marked as answer]
@alex-zwingli on ts you should use `create<T>()(...)` 

here you go -> https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgYygUwIYzXAvnAMyghDgCIAvAVwGcYMA7AEzIG4AoUSWROMNKDWB08hYqUq16zAPQhgTJgBs0Adwzo27djACe-OAGV62OAF5E7OHAAWwAOY2lDm9iYAVNAA8YALjh0UMAM9gDaALoc1naOzo5u7lAYTGgAChowDAIAkkz+gcFhkVYBNhCqhlQARukwNv5VEBAqjFG2LnGu7hBMGLoNTS0MHLgcOvo4AILIMMAQDOaW1jRoMAASHS4J3n5wABTYPvkwQSERAJTmAHxwAG4QCm0r65vxaB5JKemwWVC5-gdPmkMr9-gEToULtc7g8mE9VoYyhVqrV6vsaEiBs1MAxLmYbvdHiVnhtYltur1+vsYk4tlihniCbCRmNvNx4Mh5iJaGhDMgbO8qCpDKtZiEaItUJhsAAeYxYHAAMjg01m8yue3OexK-EEwhg2us1j2z0Z+wQJSN7TJbw8O38EQANJajSTXq53p4jvtDjAzc89kgaZ1tt7fXhzs6XdF3Qkgd9Mjk8nAndGAqtSbTbYlksCfkmASdcwnQUxGWnlqtA9asx6PsWQQW4EWvo2-kwI1GrcskZUalg0QQMEoVs7u+mYIjyn3UQCMeV-VWkPPkf26vkkZ3tOPg+Sen1-EOR2gx923TauvuqXtd-FFwag7GKQeayGt1bcJGXRbxwwMCA0H8MgaH5QUVAAWmeMV7BoMhHTgGQZDgKgGGAABHKgcD-ADCGgZsBTgJQIGQYdjGgDB7BwABrNBdBdXASnOdhzlYIA
